src/views/SystemIntake/ContactDetails/index.tsx

```diff
@@ -144,23 +128,87 @@ const ContactDetails = ({ systemIntake }: ContactDetailsProps) => {
     return link;
   })();
 
-  const onSubmit = (values?: ContactDetailsForm) => {
-    if (values) {
+  const onSubmit = async (
+    values: ContactDetailsForm,
+    { setFieldValue }: FormikHelpers<ContactDetailsForm>
+  ) => {
+    const updateSystemIntakeContact = async (type: SystemIntakeRoleKeys) => {
+      if (values[type].euaUserId && values[type].component) {
+        if (values?.[type].id) {
+          return updateContact({ ...values[type] });
+        }
+        return createContact(values[type]).then(newContact => {
+          setFieldValue(`${type}.id`, newContact?.id);
+        });
+      }
+      return null;
+    };
+
+    return Promise.all([
+      updateSystemIntakeContact('businessOwner'),
+      updateSystemIntakeContact('productManager'),
+      updateSystemIntakeContact('isso')
+    ]).then(() =>
       mutate({
         variables: {
           input: {
             id,
-            ...values,
+            requester: {
+              name: values.requester.name,
+              component: values.requester.component
+            },
+            businessOwner: {
+              name: values.businessOwner.commonName,
+              component: values.businessOwner.component
+            },
+            productManager: {
+              name: values.productManager.commonName,
+              component: values.productManager.component
+            },
+            isso: {
+              isPresent: values.isso.isPresent,
+              name: values.isso.commonName
+            },
             governanceTeams: values.governanceTeams || []
           }
         }
+      })
+    );
+  };
+
+  useEffect(() => {
+    if (authState?.isAuthenticated && !intakeRequester.euaUserId) {
+      oktaAuth.getUser().then((info: any) => {
+        setIntakeRequester({
+          commonName: info.name,
+          euaUserId: systemIntake.euaUserId,
+          email: info?.email
+        });
       });
     }
-  };
+  }, [authState, oktaAuth, systemIntake.euaUserId, intakeRequester]);
+
+  useEffect(() => {
+    if (
+      isReqAndBusOwnerSame === null &&
+      intakeRequester.euaUserId === contacts?.businessOwner.euaUserId
+    ) {
+      setReqAndBusOwnerSame(true);
+    }
+    if (
+      isReqAndProductManagerSame === null &&
+      intakeRequester.euaUserId === contacts?.productManager.euaUserId
+    ) {
+      setReqAndProductManagerSame(true);
+    }
+  }, [contacts, intakeRequester]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Removing the null check on the now `false` boolean will solve the exhaustive depth issue.

```
useEffect(() => {
    if (intakeRequester.euaUserId === contacts?.businessOwner.euaUserId) {
      setReqAndBusOwnerSame(true);
    }
    if (intakeRequester.euaUserId === contacts?.productManager.euaUserId) {
      setReqAndProductManagerSame(true);
    }
  }, [contacts, intakeRequester]);
```